pkg/reconciler/mtnamespace/controller.go

```diff
@@ -46,19 +51,40 @@ func NewController(
 	cmw configmap.Watcher,
 ) *controller.Impl {
 
+	logger := logging.FromContext(ctx)
 	namespaceInformer := namespace.Get(ctx)
 	brokerInformer := broker.Get(ctx)
 
+	recorder := controller.GetEventRecorder(ctx)
+	if recorder == nil {
+		// Create event broadcaster
+		logger.Debug("Creating event broadcaster")
+		eventBroadcaster := record.NewBroadcaster()
+		watches := []watch.Interface{
+			eventBroadcaster.StartLogging(logger.Named("event-broadcaster").Infof),
+			eventBroadcaster.StartRecordingToSink(
+				&v1.EventSinkImpl{Interface: client.Get(ctx).CoreV1().Events("")}),
+		}
+		recorder = eventBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: controllerAgentName})
+		go func() {
+			<-ctx.Done()
+			for _, w := range watches {
+				w.Stop()
+			}
+		}()
+	}
+
 	r := &Reconciler{
-		Base:            reconciler.NewBase(ctx, controllerAgentName, cmw),
-		namespaceLister: namespaceInformer.Lister(),
-		brokerLister:    brokerInformer.Lister(),
+		eventingClientSet: eventingclient.Get(ctx),
+		namespaceLister:   namespaceInformer.Lister(),
+		brokerLister:      brokerInformer.Lister(),
+		recorder:          recorder,
```

Comment:
the stuff inside of `ReconcileKind` should just be fetching `recorder` off of `ctx`

Reply:
this is a k8s type func (r *Reconciler) Reconcile(ctx context.Context, key string) error {